Fabric.Authorization.API/Modules/UsersModule.cs

```diff
@@ -37,31 +38,31 @@ public class UsersModule : FabricModule<User>
             Get("/permissions", async _ => await GetCurrentUserPermissions().ConfigureAwait(false), null,
                 "GetUserPermissions");
 
-            Post("/{userId}/AdditionalPermissions",
+            Post("/{subjectId}/AdditionalPermissions",
                 async param => await this.AddGranularPermissions(param, denied: false).ConfigureAwait(false), null,
                 "AddPermissions");
 
-            Post("/{userId}/DeniedPermissions",
+            Post("/{subjectId}/DeniedPermissions",
                 async param => await this.AddGranularPermissions(param, denied: true).ConfigureAwait(false), null,
                 "AddDeniedPermissions");
 
-            Get("/{subjectId}/groups", async _ => await GetUserGroups().ConfigureAwait(false), null, "GetUserGroups");
+            Get("/{subjectId}/{identityProvider}/groups", async _ => await GetUserGroups().ConfigureAwait(false), null, "GetUserGroups");
         }
 
         private async Task<dynamic> GetUserGroups()
         {
             this.RequiresClaims(AuthorizationReadClaim);
             var groupUserRequest = this.Bind<GroupUserRequest>();
-            var groups = await _userService.GetGroupsForUser(groupUserRequest.SubjectId);
+            var groups = await _userService.GetGroupsForUser(groupUserRequest.SubjectId, groupUserRequest.IdentityProvider);
             return groups;
         }
 
         private async Task<dynamic> AddGranularPermissions(dynamic param, bool denied)
         {
             var apiModel = this.Bind<GranularPermissionApiModel>();
 
-            if (apiModel.Target != param["userId"])
-                return CreateFailureResponse("Target must be the user id.", HttpStatusCode.BadRequest);
+            if (apiModel.Target != param["subjectId"])

```

Comment:
Can we make this a case insensitive comparison? Also - not related to your changes, I am wondering why we make this check, i.e. what is the purpose of the Target property on the GranularPermissionApiModel? @deeiinu can you refresh my memory on the purpose of the Target property?

Reply:
I was curious about that myself. I realize that it represents the subjectId of the user to change permissions for but I am not sure why its called Target.